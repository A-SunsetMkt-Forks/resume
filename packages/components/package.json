{
  "name": "@resumejs/components",
  "version": "0.0.8",
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    },
    "./style": {
      "import": "./dist/style.css"
    }
  },
  "scripts": {
    "dev": "vite build --watch",
    "build": "vite build --emptyOutDir",
    "prepublishOnly": "pnpm run build"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "dependencies": {
    "@resumejs/template": "workspace:*",
    "clsx": "^1.1.1",
    "hast": "^1.0.0",
    "hast-util-has-property": "^2.0.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-markdown": "^8.0.1",
    "rehype-raw": "^6.1.1",
    "remark-frontmatter": "^4.0.1",
    "remark-gfm": "^3.0.1",
    "unified": "^10.1.2",
    "unist-builder": "^3.0.0",
    "unist-util-is": "^5.1.1",
    "unist-util-visit": "^4.1.0",
    "vite-plugin-dts": "^1.1.1",
    "yaml": "^2.0.1"
  },
  "devDependencies": {
    "@types/react": "^17.0.33",
    "@types/react-dom": "^17.0.10",
    "@vitejs/plugin-react": "^1.0.7",
    "vite": "^2.9.0"
  }
}
