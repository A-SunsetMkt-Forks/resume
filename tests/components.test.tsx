import { describe, expect, it } from 'vitest'
import { Resume } from '@resumejs/components'
import { render } from '@testing-library/react'
import React from 'react'

describe('components', () => {
  const md = `
    # 张三

    ![头像](https://notion-avatar.vercel.app/api/svg/eyJmYWNlIjo5LCJub3NlIjoxMCwibW91dGgiOjEsImV5ZXMiOjcsImV5ZWJyb3dzIjoxMSwiZ2xhc3NlcyI6MCwiaGFpciI6MTEsImFjY2Vzc29yaWVzIjoxMywiZGV0YWlscyI6MCwiYmVhcmQiOjAsImZsaXAiOjAsImNvbG9yIjoicmdiYSgyNTUsIDAsIDAsIDApIiwic2hhcGUiOiJub25lIn0=)

    - [https://github.com/Dunqing](https://github.com/Dunqing)

    >

    - google@gmail.com
    - 3 年

    >

    - Telegram: @luckingforme
  `
  it('heading', () => {
    const { container } = render(<Resume components={{}}>{md}</Resume>)
    console.log(container.innerHTML)
    expect(container.innerHTML).toMatchSnapshot()
  })
})
