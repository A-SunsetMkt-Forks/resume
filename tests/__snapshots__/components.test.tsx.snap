// Vitest Snapshot v1

exports[`components > heading 1`] = `
"<div class=\\"r-resume\\"><div class=\\"font-sans text-base text-dark-400 bg-white dark:text-light-700 dark:bg-dark-400 r-container\\"><div class=\\"border border-light-700 md:w-full md:max-w-screen-md xs:w-full mx-auto p-y-5 sm:p-x-10 p-x-4 dark:border-dark-100 relative r-content\\"><div class=\\"absolute right-4 top-2 flex gap-x-1.5 print:hidden r-toolbox\\"></div><pre class=\\"bg-light p-x-5 p-y-3 dark:bg-dark-300 r-pre\\"><code class=\\"p-x-1.5 p-y-0.5 bg-neutral-100 rounded-md text-xs dark:bg-neutral-500 r-code\\"># 张三

![头像](https://notion-avatar.vercel.app/api/svg/eyJmYWNlIjo5LCJub3NlIjoxMCwibW91dGgiOjEsImV5ZXMiOjcsImV5ZWJyb3dzIjoxMSwiZ2xhc3NlcyI6MCwiaGFpciI6MTEsImFjY2Vzc29yaWVzIjoxMywiZGV0YWlscyI6MCwiYmVhcmQiOjAsImZsaXAiOjAsImNvbG9yIjoicmdiYSgyNTUsIDAsIDAsIDApIiwic2hhcGUiOiJub25lIn0=)

- [https://github.com/Dunqing](https://github.com/Dunqing)

&gt;

- google@gmail.com
- 3 年

&gt;

- Telegram: @luckingforme
</code></pre></div></div></div>"
`;
